### ISSUE TITLE
Incorrect completion when \n and \t used in file names

### ISSUE THREAD

**Issue by xzfc** [fish-shell/fish-shell#1537]
Example:

```
mkdir 1\n2 a\tb
cd <TAB> # completes `1`, `2/`, and `a`
ls <TAB> # completes `1\n2/`, and `a`
```

Both variants should complete `1\n2/` and `a\tb/`

**Comment by krader1961** [comment]
I just confirmed that this is still suboptimal using fish built from git head. Obviously file name completion needs more escaping.

**Comment by mqudsi** [comment]
This is fixed in fish 3.0, (probably with the fixes to `__fish_complete_files`).

See also https://github.com/fish-shell/fish-shell/issues/4789
